i2c/bus_scan/bus_scan.c

```diff
@@ -33,13 +34,18 @@ bool reserved_addr(uint8_t addr) {
 int main() {
     // Enable UART so we can print status output
     stdio_init_all();
-
-    // This example will use I2C0 on GPIO4 (SDA) and GPIO5 (SCL)
-    i2c_init(i2c0, 100 * 1000);
-    gpio_set_function(4, GPIO_FUNC_I2C);
-    gpio_set_function(5, GPIO_FUNC_I2C);
-    gpio_pull_up(4);
-    gpio_pull_up(5);
+#if !defined(i2c_default) || !defined(PICO_DEFAULT_I2C_SDA_PIN) || !defined(PICO_DEFAULT_I2C_SCL_PIN)
+#warning i2c/bus_scane example requires a board with I2C pins
+    puts("Default I2C pins were not defined");
+#else
+    // This example will use I2C0 on the default SDA and SCL pins (4, 5 on a Pico)
+    i2c_init(i2c_default, 100 * 1000);
+    gpio_set_function(PICO_DEFAULT_I2C_SDA_PIN, GPIO_FUNC_I2C);
+    gpio_set_function(PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C);
+    gpio_pull_up(PICO_DEFAULT_I2C_SDA_PIN);
+    gpio_pull_up(PICO_DEFAULT_I2C_SCL_PIN);
+    // Make the I2C pins available to picotool
+    bi_decl(bi_2pins_with_func(PICO_DEFAULT_I2C_SDA_PIN, PICO_DEFAULT_I2C_SCL_PIN, GPIO_FUNC_I2C));
```

Comment:
Would this PR be a good place to add `bi_decl` stuff to all the other examples too, or would that make more sense in a separate PR?

Reply:
separate PR - feel free :-)

Reply:
LOL, I'm not that familiar with the binary_info stuff yet... maybe towards the end of the week :wink: